config/confighttp/confighttp.go

```diff
@@ -99,8 +100,31 @@ func NewDefaultHTTPClientSettings() HTTPClientSettings {
 	}
 }
 
+// toClientOptions has options that change the behavior of the HTTP client
+// returned by HTTPClientSettings.ToClient().
+type toClientOptions struct {
+	customRoundTripper func(next http.RoundTripper) (http.RoundTripper, error)
+}
+
+// ToClientOption is an option to change the behavior of the HTTP client
+// returned by HTTPClientSettings.ToClient().
+type ToClientOption func(opts *toClientOptions)
```

Comment:
I think we decided to go with the model of having Options as interface with a private func `apply`

Reply:
I also updated `ToServerOption` to make the code consistent.